### DISCUSSION TITLE
Is there a way to publish artifacts but disable the release pipeline dynamically?

### DISCUSSION THREAD

**Discussion by timflannagan** [goreleaser/goreleaser#3042] [Q&A] [answered]
Awesome tool - I've used this in a couple of different projects already and I've been super happy integrating it in CI/CD pipelines.

Hopefully this is a quick question: I was looking into consolidating our deploy/release pipeline and wanted to use goreleaser to build and publish container and manifestlist images when new tags are created, and when new commits merge to the default branch (e.g. main).

> Note: When I was poking at the documentation, I didn't see a CLI flag for configuring this behavior directly, and the --skip-publish seems to disable both the publish and release pipeline.

For the latter case, I'm interested in publishing those goreleaser built container images, but I'd like to avoid have a release published, which I think gives me the following options:

- Use a different .goreleaser.yaml for tag events vs. main branch push events, and override that configuration file at CI runtime.
- Disable the release pipeline for both event triggers, and manually publish release artifacts using GHA workflows.
- Dynamically flip the release.disable value to true/false.

I took a stab at trying to dynamically flipping this `release.disable` field, which feels like the hackiest solution by far but I wanted to give it a shot, and I quickly ran into YAML unmarshaling errors as that field expects a boolean value, and I wasn't able to get something like `{{ .Env.DISABLE_RELEASE_PIPELINE }}` to work without quoting the template.

**Comment by caarlos0** [comment, marked as answer]
Hey, this is possible as a pro feature

https://goreleaser.com/customization/nightlies/

You could also use envsubst or something like that to replace things in the yaml before running goreleaser.

hope this helps :)

**Reply by timflannagan** [reply]
Thanks!
